src/commands/deploy/setPreDeployTaskForDotnet.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import { QuickPickItem, TextDocument, workspace, WorkspaceFolder } from 'vscode';
+import { ext } from 'vscode-azureappservice/out/src/extensionVariables';
+import * as constants from '../../constants';
+import { nonNullProp } from '../../utils/nonNull';
+import * as workspaceUtil from '../../utils/workspace';
+import { getWorkspaceSetting, updateWorkspaceSetting } from '../../vsCodeConfig/settings';
+import * as tasks from '../../vsCodeConfig/tasks';
+import { IDeployWizardContext } from "../createWebApp/setAppWizardContextDefault";
+
+export async function setPreDeployTaskForDotnet(context: IDeployWizardContext): Promise<void> {
+    const fsPath: string = nonNullProp(context, 'fsPath');
+    const currentWorkspace: WorkspaceFolder | undefined = workspaceUtil.getContainingWorkspace(fsPath);
+    // follow the publish output patterns, but leaveout tfw
+    const dotnetOutputPath: string = path.join('bin', 'Debug', 'publish');
+
+    // tslint:disable-next-line: strict-boolean-expressions
+    if (!currentWorkspace || !getWorkspaceSetting('configurePreDeployTasks', currentWorkspace.uri.fsPath)) {
+        // if the workspace being deployed is not opened, return and do nothing
+        return;
+    }
+
+    const csProj = await workspaceUtil.findFilesByFileExtension(currentWorkspace.uri.fsPath, 'csproj');
+    if (csProj.length > 0) {
```

Comment:
Would it be better to check the siteConfig.linuxFxVersion is .NET? Or both?

Reply:
Good question - I'd have to think about it more. Can you explain why you chose what you did?

Reply:
I think the .csproj file is what's use to detect whether or not something is a C# project in the Omnisharp extension @fiveisprime at least recommended that's what we do.  

A publish will fail if we don't have the .csproj file, so I guess it makes sense to check for that.

Reply:
Omnisharp is a completely different scenario IMO. They _add_ .NET functionality to your workspace, but don't block other languages. Here we have to choose one and _only_ one language used for deploying.

We already have logic to detect if a project is a node.js app (aka if it has a package.json) or java app. I think you need to somehow refactor so that if a workspace matches multiple languages (aka both a csproj and a package.json) you can bail out and not use the defaults for either language.

As an alternative, perhaps you _should_ check `siteConfig.linuxFxVersion` since that seems more deterministic. But that would only work for linux and might have other drawbacks